### PR TITLE
Offline Activities Model and API

### PR DESCRIPTION
**Pull request by RotaruDan**

### CODE REVIEW

**Comment by RotaruDan** [comment]
This is a PR to the branch 3in1 https://github.com/e-ucm/rage-analytics-backend/pull/132

**Comment by RotaruDan** [comment]
This is a PR to the branch 3in1 https://github.com/e-ucm/rage-analytics-backend/pull/132

File: test/tests/offlinetracesTest.js [hunk 1]
```diff
@@ -169,6 +169,7 @@ module.exports = function (request, db) {
                 .end(function (err, res) {
                     console.error(err);
                     console.info(res.body);
+                    console.info(res);
```
Comment (Synpheros) [inline test/tests/offlinetracesTest.js]: Probably remove this?

File: lib/offlinetraces.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                    return yield sendTraces();
+                });
+                /*.catch(function (err) {
+                    // Log any uncaught errors
+                    // co will not throw any errors you do not handle!!!
+                    // HANDLE ALL YOUR ERRORS!!!
+                    console.log('Error while sending traces to collector!');
+                    console.error(err);
+                    return err;
+                });*/
```
[the remark this answers is not in the document]
Comment (RotaruDan) [reply lib/offlinetraces.js]: Yes!

File: .travis.yml [hunk 3]
```diff
@@ -10,8 +10,10 @@ addons:
 
 services:
   - mongodb
+  - docker
 
 before_install:
+  - docker run -p 9092:9092 -p 2181:2181 --env ADVERTISED_HOST=localhost --env ADVERTISED_PORT=9092 -d eucm/kzk:0.11.0.1
```
[the remark this answers is not in the document]
Comment (RotaruDan) [reply .travis.yml]: True

File: lib/offlinetraces.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                console.log('result');
+                console.info(result);
+
+
+                var sendTraces = function* () {
+
+                    var users = {};
+                    var data = fs.readFileSync(req.file.path, 'utf8');
+
+                    var csvtraces = data.split('\n');
```
[the remark this answers is not in the document]
Comment (RotaruDan) [reply lib/offlinetraces.js]: True

File: lib/offlinetraces.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+    /**
+     * Creates a new offlinetraces for the given activityId.
+     * @Returns a promise with the analysis created
+     */
+    offlinetraces.createOfflinetraces = function (activityId, username, req, res) {
+
+        activityId = offlinetraces.toObjectID(activityId);
+        var trackingCode = '';
+        return activities.findById(activityId)
```
[the remark this answers is not in the document]
Comment (RotaruDan) [reply lib/offlinetraces.js]: Yep

File: test/tests/offlinetracesTest.js (hunk 1 above)
Comment (RotaruDan) [reply test/tests/offlinetracesTest.js]: yes

**Comment by RotaruDan** [comment]
All comments addressed

**Comment by RotaruDan** [comment]
Closing and reopening to master

**Pull request closed without merging**